### PR TITLE
Suppress mock_mode for download of intermediate files

### LINKED ISSUE #57: Mock_mode prevents downloads of intermediate files
When mock mode is used, intermediate files cannot be downloaded because the docker call that downloads them is suppressed.

### PR DESCRIPTION
**Pull request by almussel**
Resolves #57

### CODE REVIEW

File: src/toil_lib/urls.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 _log = logging.getLogger(__name__)
 
 
-def download_url(job, url, work_dir='.', name=None, s3_key_path=None, cghub_key_path=None):
+def download_url(job, url, work_dir='.', name=None, s3_key_path=None, cghub_key_path=None, mock=None):
```
Comment (hannes-ucsc) [inline src/toil_lib/urls.py]: Document in docstring. It's ok to refer to `docker_call` with 
```
See :func:`toil_lib.programs.docker_call`.
```

**Comment by almussel** [comment]
@hannes-ucsc I've made the requested changes.

File: src/toil_lib/urls.py [hunk 2]
```diff
@@ -23,14 +23,15 @@ def download_url(job, url, work_dir='.', name=None, s3_key_path=None, cghub_key_
     :param str name: Name of output file, if None, basename of URL is used
     :param str s3_key_path: Path to 32-byte encryption key if url points to S3 file that uses SSE-C
     :param str cghub_key_path: Path to cghub key used to download from CGHub.
+    :param bool mock: Whether to run docker calls with mock_mode. See :func:`toil_lib.programs.docker_call`
```
Comment (hannes-ucsc) [inline src/toil_lib/urls.py]: in mock mode

File: src/toil_lib/urls.py [hunk 3]
```diff
@@ -94,6 +95,7 @@ def _s3am_with_retry(job, num_cores, file_path, s3_url, mode='upload', s3_key_pa
     :param str s3_url: S3 URL
     :param str mode: Mode to run s3am in. Either "upload" or "download"
     :param str s3_key_path: Path to the SSE-C key if using encryption
+    :param bool mock: Whether to run docker calls with mock_mode. See :func:`toil_lib.programs.docker_call`
```
Comment (hannes-ucsc) [inline src/toil_lib/urls.py]: in mock mode

File: src/toil_lib/urls.py (hunk 2 above)
Comment (almussel) [inline src/toil_lib/urls.py]: That's actually not true. If `mock` is set to `True`,  `docker_call` will run in mock mode regardless of whether the mock mode environment variable is set. Whether that functionality actually makes sense is debatable, though.

Comment (hannes-ucsc) [inline src/toil_lib/urls.py]: I was just asking for a minor spelling fix. `with mock_mode` is a bit awkward. I thought `in mock mode` sounded better. The semantics of this parameter is the same as for docker_call, correct? True enables mock mode, False disables it and None (the default) uses the environment variable or, if that's not set, ultimately disables it. Right?

Comment (almussel) [inline src/toil_lib/urls.py]: Yes, that's correct. I misunderstood what you were asking for. I will change the wording.

**Comment by almussel** [comment]
@hannes-ucsc fixed!

**Pull request merged**